l3kernel/l3keys.dtx

```diff
@@ -2371,17 +2370,17 @@
 %     \@@_set:nn
 %   }
 % \begin{macro}{\@@_set:nnn}
-%   A simple wrapper allowing for nesting.
+%   A simple wrapper allowing for nesting. The first argument to |\@@_set:nnn|
+%   should always be a string.
 %    \begin{macrocode}
 \cs_new_protected:Npn \keys_set:nn #1#2
   {
-    \use:x
+    \use:e
```

Comment:
Why?  `\use:e` is faster in recent engines but extremely slow otherwise, and this is not worth a separate code branch depending on whether `\expanded` is available, because it is not in the "inner loop" of setting each individual key.